vdr/types/interface.go

```diff
@@ -81,31 +81,27 @@ type DocUpdater interface {
 // KeyResolver is the interface for resolving keys.
 // This can be used for checking if a signing key is valid at a point in time or to just find a valid key for signing.
 type KeyResolver interface {
-	// ResolveSigningKeyID looks up a signing key of the specified holder. It returns the ID
-	// of the found key. Typically used to find a key for signing one's own documents. If no suitable keys
-	// are found an error is returned.
-	ResolveSigningKeyID(holder did.DID, validAt *time.Time) (string, error)
-	// ResolveSigningKey looks up a specific signing key and returns it as crypto.PublicKey. If the key can't be found
-	// or isn't meant for signing an error is returned.
-	ResolveSigningKey(keyID string, validAt *time.Time) (crypto.PublicKey, error)
-	// ResolveAssertionKeyID look for a valid assertion key for the give DID. If multiple keys are valid, the first one is returned.
-	// An ErrKeyNotFound is returned when no key is found.
-	ResolveAssertionKeyID(id did.DID) (ssi.URI, error)
-	// ResolveKeyAgreementKey look for a valid keyAgreement key for the give DID. If multiple keys are valid, the first one is returned.
-	// An ErrKeyNotFound is returned when no key is found.
-	ResolveKeyAgreementKey(id did.DID) (crypto.PublicKey, error)
+	// ResolveKeyByID looks up a specific key of the given RelationType and returns it as crypto.PublicKey.
+	// If multiple keys are valid, the first one is returned.
+	// An ErrKeyNotFound is returned when no key (of the specified type) is found.
+	ResolveKeyByID(keyID string, validAt *time.Time, relationType RelationType) (crypto.PublicKey, error)
```

Comment:
I think it would unnecessarily complicate the usage of these functions, since we should then pass a DID URL instead of a key ID as well. But I think we could introduce a dereferencer later, between the KeyResolver and DIDResolver?

Maybe for a future PR, his basically changes the function signatures, I'd rather change semantics in a follow-up PR?

Reply:
I incorrectly assumed that keyID was the DID URL